modules/040-node-manager/hooks/generate_static_kubeconfig_secret.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2023 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hooks
+
+import (
+	"context"
+	"fmt"
+	"github.com/deckhouse/deckhouse/go_lib/dependency/k8s"
+	v1 "k8s.io/api/core/v1"
+	"os"
+
+	"github.com/pkg/errors"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook"
+	"github.com/flant/addon-operator/sdk"
+	"github.com/flant/shell-operator/pkg/kube/object_patch"
+	"github.com/flant/shell-operator/pkg/kube_events_manager/types"
+
+	"github.com/deckhouse/deckhouse/go_lib/dependency"
+	"github.com/deckhouse/deckhouse/modules/040-node-manager/hooks/internal/kubeconfig"
+)
+
+const (
+	clusterAPINamespace          = "d8-cloud-instance-manager"
+	clusterAPIServiceAccountName = "capi-controller-manager"
+)
+
+var _ = sdk.RegisterFunc(&go_hook.HookConfig{
+	Kubernetes: []go_hook.KubernetesConfig{
+		{
+			Name:       "cluster_api_cluster",
+			ApiVersion: "cluster.x-k8s.io/v1beta1",
+			Kind:       "Cluster",
+			NamespaceSelector: &types.NamespaceSelector{
+				NameSelector: &types.NameSelector{
+					MatchNames: []string{clusterAPINamespace},
+				},
+			},
+			FilterFunc: applyClusterFilter,
+		},
+	},
+}, dependency.WithExternalDependencies(generateStaticKubeconfigSecret))
+
+func applyClusterFilter(obj *unstructured.Unstructured) (go_hook.FilterResult, error) {
+	return obj, nil
```

Comment:
return the whole obj isn't good, because obj resourceVersion is periodically changed and hook will be started ion every change

Reply:
I think we should rewrite hook on cron basis, and get cluster from api by k8s client

Reply:
done